preditor/gui/loggerwindow.py

```diff
@@ -539,41 +544,72 @@ def handleMenuHovered(self, action):
         menu = action.parentWidget()
         QToolTip.showText(QCursor.pos(), text, menu)
 
-    def findCurrentFontAction(self):
-        """Find and return current font's action"""
-        actions = self.uiMonospaceFontMENU.actions()
-        actions.extend(self.uiProportionalFontMENU.actions())
-
-        action = None
-        for act in actions:
-            if act.isChecked():
-                action = act
-                break
-
-        return action
-
-    def selectFont(self, action):
-        """Set console and workbox font to current font
+    def selectFont(self, monospace=False, proportional=False):
+        """Present a QFontChooser dialog, offering, monospace, proportional, or all
+        fonts, based on user choice. If a font is chosen, set it on the console and
+        workboxes.
 
         Args:
-            action (QAction): menu action associated with chosen font
+            action (QAction): menu action associated with chosenO font
         """
+        origFont = self.console().font()
+        curFontFamily = origFont.family()
+
+        if monospace and proportional:
+            options = QFontDialog.MonospacedFonts | QFontDialog.ProportionalFonts
+            kind = "monospace or proportional "
+        elif monospace:
+            options = QFontDialog.MonospacedFonts
+            kind = "monospace "
+        elif proportional:
+            options = QFontDialog.ProportionalFonts
+            kind = "proportional "
+
+        # Present a QFontDialog for user to choose a font
+        title = "Pick a {} font. Current font is: {}".format(kind, curFontFamily)
+        newFont, okClicked = QFontDialog.getFont(
+            origFont, self, title, options=options
+        )
 
-        actions = self.uiMonospaceFontMENU.actions()
-        actions.extend(self.uiProportionalFontMENU.actions())
+        if okClicked:
+            self.console().setConsoleFont(newFont)
+            self.setWorkboxFontBasedOnConsole()
+            self.setEditorChooserFontBasedOnConsole()
 
-        for act in actions:
-            act.setChecked(act == action)
+    def setFontSize(self, newSize):
+        """Update the font size in the console and current workbox.
 
-        family = action.text()
+        Args:
+            newSize (int): The new size to set the font
+        """
         font = self.console().font()
-        font.setFamily(family)
+        font.setPointSize(newSize)
         self.console().setConsoleFont(font)
 
-        for workbox in self.uiWorkboxTAB.all_widgets():
+        self.setWorkboxFontBasedOnConsole()
+        self.setEditorChooserFontBasedOnConsole()
+
+    def setWorkboxFontBasedOnConsole(self):
+        """If the current workbox's font is different to the console's font, set it to
+        match.
+        """
+        font = self.console().font()
+        workbox = self.uiWorkboxTAB.currentWidget().currentWidget()
+        if workbox is None:
+            return
+        if workbox.font != font:
```

Comment:
Ahh, yes. Not only that, it should call `workbox.__font__()`, otherwise it's apparently the `Qscintilla.font()` method, and will not use the lexer font when available.